### DISCUSSION TITLE
DataGrid 相关绑定问题

### DISCUSSION THREAD

**Discussion by houlongfei** [BYJRK/DotNet-Discussions#204] [MVVM]
我做如下一个view
![image](https://github.com/BYJRK/DotNet-Discussions/assets/15973049/58c9c0ea-67eb-4b6e-8e85-e4b67ec1c6a8)

区域一中DataGrid绑定一个   ObservableCollectio类型
区域二绑定的是区域一DataGrid的选中行

现在需求，区域二中的值修改后，需要通过点击“确认“按键 ，才能更新区域一的表格。但现在的情况是，只要修改区域二的TextBox的值，表格中的值也就立即更新了。

有好的建议和方法？

**Comment by RickyHe7** [comment]
区域2里的值就不要绑定了，把里面的选中行做成事件，选中后做区域2的更新，改完值后点确定再更改observableCollection里的值通知，不知道可不可行

**Comment by TRARS** [comment]
[UpdateSourceTrigger.Explicit](https://learn.microsoft.com/zh-cn/dotnet/desktop/wpf/data/how-to-control-when-the-textbox-text-updates-the-source?view=netframeworkdesktop-4.8)

**Reply by houlongfei** [reply]
区域2要是有大量的输入框呢，每个都Explicit?

**Reply by TRARS** [reply]
框多那换个法子，单向绑定左至右。
左边数据源：ObservableCollection&lt;UserViewModel&gt; UserListVM;
右边数据源：UserViewModel UserTempVM;
确定按钮的Command中实现：UserTempVM.CopyTo(被选中行VM);  （CopyTo自己实现）

**Reply by houlongfei** [reply]
目前是这么做的。
